test/lazy_html_test.exs

```diff
@@ -250,6 +250,134 @@ defmodule LazyHTMLTest do
     end
   end
 
+  describe "parent_node/1" do
+    test "from selector of nodes on different levels" do
+      lazy_html =
+        LazyHTML.from_fragment("""
+        <div id="a">
+          <div id="b">
+            <span>Hello</span>
+          </div>
+          <span>world</span>
+        </div>
+        """)
+
+      spans = LazyHTML.query(lazy_html, "span")
+      parents = LazyHTML.parent_node(spans)
+      parent_ids = parents |> Enum.flat_map(&LazyHTML.attribute(&1, "id")) |> Enum.sort()
+      assert parent_ids == ["a", "b"]
+
+      # parent of div#id="a" is null
+      grandparents = LazyHTML.parent_node(parents)
+      assert LazyHTML.tag(grandparents) == ["div"]
+
+      great_grandparents = LazyHTML.parent_node(grandparents)
+      assert great_grandparents |> Enum.count() == 0
+    end
+
+    test "from selector of nodes on same level" do
+      lazy_html =
+        LazyHTML.from_fragment("""
+        <div id="a">
+          <div id="b">
+            <span>Hello</span>
+          </div>
+          <div id="c">
+            <span>world</span>
+          </div>
+        </div>
+        """)
+
+      spans = LazyHTML.query(lazy_html, "span")
+      parents = LazyHTML.parent_node(spans)
+      parent_ids = parents |> Enum.flat_map(&LazyHTML.attribute(&1, "id")) |> Enum.sort()
+      assert parent_ids == ["b", "c"]
+
+      # since they share the same parent, we now only have one node left
+      grandparent = LazyHTML.parent_node(parents)
+      assert LazyHTML.attribute(grandparent, "id") == ["a"]
+    end
+
+    defp parents(node) do
+      if Enum.count(node) == 0 do
+        []
+      else
+        tag = LazyHTML.tag(node)
+        parents(LazyHTML.parent_node(node)) ++ tag
+      end
+    end
+
+    test "last parent node is <html> if instantiated via from_document and similar" do
+      lazy_html = LazyHTML.from_document("<html><body><div>root</div></body></html>")
+      assert parents(lazy_html["div"]) == ["html", "body", "div"]
```

Comment:
Nit: I would avoid using access protocol here to reduce coupling the test with other features:

```suggestion
      assert parents(LazyHTML.query(lazy_html, "div")) == ["html", "body", "div"]
```

(same below)